src/mongo/mongoConnectionStrings.ts

```diff
@@ -3,19 +3,55 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-export function getDatabaseNameFromConnectionString(connectionString: string): string | undefined {
-    //  Connection strings follow the following format (https://docs.mongodb.com/manual/reference/connection-string/):
-    // mongodb://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
+function parseConnectionString(connectionString: string): string[] | undefined {
+    // Connection strings follow the following format (https://docs.mongodb.com/manual/reference/connection-string/):
+    //   mongodb[+srv]://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
     // Some example connection strings:
-    //   mongodb://dbuser:dbpassword@dbname.mlab.com:14118/dbname
-    //   mongodb://db1.example.net:27017,db2.example.net:2500/?replicaSet=test
-    //   mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/
+    //   mongodb://dbuser:dbpassword@dbname.mlab.com:14118
+    //   mongodb+srv://db1.example.net:27017,db2.example.net:2500/?replicaSet=test
+    //   mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/database?ssh=true
+    // Regex splits into five parts:
+    //   Full match
+    //   mongodb[+srv]
+    //   [username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]]
+    //   [database]
+    //   [?options]
+
+    try {
+        let matches = connectionString.match('(^mongodb(?:[+]srv)?):\/\/([^\/]*)(?:\/([^/?]*)(.*))?$');
```

Comment:
Would be much easier to read if you break up the regex into multiple string constants.